woocommerce/woocommerce-android
# Shipping refund: Add shipping lines selection in order refund selection screen.

This is step 3 out of 4 for this feature development, [as outlined here.](https://github.com/woocommerce/woocommerce-android/issues/3319#issuecomment-783276159)

This PR deals with showing the toggle that allows user to select shipping line(s) to be refunded.

To test this, please prepare three different Orders:
1. An order with no shipping line items.
2. An order with one shipping line item.
3. An order with more than one shipping line items. This can be created in core by editing an Order, then click "Add Item(s)", then add a Shipping line item from there.

## To test:

Go to **Order list**, select an **Order**, then tap **"Issue Refund"**. 

| Case                        | Before | With this PR: step 1 | With this PR: step 2 |
|-----------------------------|--------|-------|-------|
| Order with no shipping lines        |  ![order-no-shipping-lines](https://user-images.githubusercontent.com/266376/110299851-ccec0880-8028-11eb-9dc8-84fa8e08b95e.png)  | Should stay the same as the "Before" state, no shipping refund switch shown. | Should stay the same as the "Before" state, no shipping refund switch shown.  |
| Order with 1 shipping line  |   As above, main switch ("Refund shipping") should not be shown.   |  ![order-1-shipping-line-01](https://user-images.githubusercontent.com/266376/110299800-b776de80-8028-11eb-8ef1-48cff1eb0069.png) |   ![order-1-shipping-line-02](https://user-images.githubusercontent.com/266376/110299774-af1ea380-8028-11eb-9372-32d9adbe0106.png)  |
|  |  | Should show main switch.  |  Should only show the main switch, and no individual switch next to the shipping line. |
| Order with 2 shipping lines |    As above, main switch ("Refund shipping") should not be shown.   | ![order-2-shipping-lines-01](https://user-images.githubusercontent.com/266376/110299696-93b39880-8028-11eb-80a3-ae839e88b4ab.png)  |  ![order-2-shipping-lines-02](https://user-images.githubusercontent.com/266376/110299642-84344f80-8028-11eb-9a9f-e1bd5daa4e0a.png)   |
|  |  | Should show main switch.  |   Should show the main switch and individual switch next to each shipping line.  |


When testing, please pay attention to:
- **Grand total** (products + shipping refund) as shown on the toolbar at the top, next to the "X" button. The number should stay correct as you toggle the shipping refund option, as well as when making configuration changes. This is the part where I could use some help because I might be missing some edge cases when counting grand total. 
- **Refund subtotal, taxes, and total** as shown below the list of shipping item. The number should stay correct as you toggle the shipping refund option, as well as when making configuration changes. 
- **Toggle selections** on Order with multiple shipping line items. Checked and un-checked items should stay checked/un-checked as you make configuration changes, and as you toggle the main refund switch.
